include/llbuild/BuildSystem/BuildExecutionQueue.h

```diff
@@ -85,6 +85,9 @@ class BuildExecutionQueue {
   /// Add a job to be executed.
   virtual void addJob(QueueJob job) = 0;
 
+  /// Cancel all jobs and subprocesses of this queue
+  virtual void cancelAllJobs() = 0;
```

Comment:
Please include documentation on the thread semantics of this (is it safe to call from multiple threads; current answer is kinda no), and also its semantics w.r.t. the addition of subsequent jobs.

Also, it might be worth noting here that subprocess will be interrupted/killed but subjobs (lambdas which do work, but don't spawn subprocesses) don't yet have a way to tell them to cancel. We should probably think about adding APIs to support that (even if it is just a query API to check if the queue has been cancelled).